src/openforms/js/components/admin/form_design/FormLogic.js

```diff
@@ -64,19 +131,25 @@ const Rule = ({jsonLogicTrigger, actions, onChange, onDelete}) => {
             </div>
 
             <div className="logic-rule__rule">
-                <Trigger name="jsonLogicTrigger" logic={jsonLogicTrigger} onChange={onChange} />
+                {
+                    isAdvanced
+                    ? (<AdvancedTrigger name="jsonLogicTrigger" logic={jsonLogicTrigger} onChange={onChange} error={errors['jsonLogicTrigger']}/>)
+                    : (<Trigger name="jsonLogicTrigger" logic={jsonLogicTrigger} onChange={onChange} />)
+                }
```

Comment:
the props you pass are essentially the same, but the component itself is different, so I'd rewrite that as such

```jsx
const TriggerComponent = isAdvanced ? AdvancedTrigger : Trigger;

return <TriggerComponent name="jsonLogicTrigger" logic={jsonLogicTrigger} onChange={onChange} error={errors.jsonLogicTrigger} />
```